packages/flutter_tools/bin/fuchsia_tester.dart

```diff
@@ -70,16 +81,37 @@ Future<Null> run(List<String> args) async {
     if (!fs.isFileSync(shellPath)) {
       throwToolExit('Cannot find Flutter shell at $shellPath');
     }
-    loader.installHook(
-      shellPath: shellPath,
-    );
+    // Put the tester shell where runTests expects it.
+    final Link testerDestLink =
+        fs.link(artifacts.getArtifactPath(Artifact.flutterTester));
+    testerDestLink.parent.createSync(recursive: true);
+    testerDestLink.createSync(shellPath);
 
     PackageMap.globalPackagesPath =
         fs.path.normalize(fs.path.absolute(argResults[_kOptionPackages]));
-    fs.currentDirectory = testDirectory;
 
-    await test.main(testArgs);
-    exit(exitCode);
+    CoverageCollector collector;
+    if (argResults['coverage']) {
+      collector = new CoverageCollector();
+    }
+
+    final int result = await runTests(
+      tests,
+      workDir: testDirectory,
+      watcher: collector,
+      enableObservatory: collector != null,
+    );
+
+    if (collector != null) {
+      // collector expects currentDirectory to be the root of the dart
+      // package (i.e. contains lib/ and test/ sub-dirs).
+      fs.currentDirectory = testDirectory.parent;
+      if (!await
+          collector.collectCoverageData(argResults[_kOptionCoveragePath]))
+        throwToolExit('Failed to collect coverage data');
+    }
+
+    exit(result);
```

Comment:
Optionally you can do the following above, and skip the exit call here.
```
exitCode = await runTests(...
```

Reply:
Done.